library/manageiq.py

```diff
@@ -87,59 +156,17 @@ def main():
         )
     )
 
-    params = module.params
-    url = params['url'] + '/api'
-    providers_url = url + '/providers'
-    username = params['username']
-    password = params['password']
-    provider_name = params['name']
-    port = params['port']
-    hostname = params['hostname']
-    token = params['token']
-
-    endpoints = [{'endpoint': {'role': 'default', 'hostname': hostname,
-                               'port': port},
-                  'authentication': {'role': 'bearer',
-                                     'auth_key': token}}]
-
-    client = MiqApi(url, (username, password))
-    providers = client.collections.providers
-    # check if provider with the same name already exists
-    provider_id = None
-    for provider in providers:
-        if provider.name == provider_name:
-            provider_id = provider.id
-    if provider_id: #provider exists
-        if update_required(client, module, providers_url, provider_id, hostname, port):
-            # updates provider with new parameters
-            try:
-                result = client.post(providers_url + '/%d' % provider_id,
-                                     action='edit',
-                                     connection_configurations=endpoints)
-                msg = "Successfuly updated %s provider" % provider_name
-                changed = True
-            except Exception as e:
-                msg = "Failed to add provider. Error: %s" % e
-                module.fail_json(msg=msg)
-        else:
-            msg = "Provider %s already exists" % provider_name
-            changed = False
-    # provider doesn't exists, adding it to manageiq
-    else:
-        try:
-            result = client.post(providers_url, type=PROVIDER_TYPE,
-                                 name=provider_name,
-                                 connection_configurations=endpoints)
-            provider_id = result['results'][0]['id']
-            msg = "Successfuly added %s provider" % provider_name
-            changed = True
-        except Exception as e:
-            msg = "Failed to add provider. Error: %s" % e
-            module.fail_json(msg=msg)
+    url           = module.params['url'] + '/api'
```

Comment:
consider calling this `api_url` to avoid different meanings between module's parameter and internal variable of same name.